### PR TITLE
SWATCH-1241 remove tag profile from tally resource

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: [SWATCH-1241](https://issues.redhat.com/browse/SWATCH-1241)

Draft until https://github.com/RedHatInsights/rhsm-subscriptions/pull/2369 merged.
## Description

Remove deprecated tag_profile and use swatch-product-configuration instead in TallyResource.

### Setup

1. Start app with : `DEV_MODE=true ./gradlew :bootRun`
1. Ensure account is opted in: 
```
curl -X 'PUT' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/opt-in' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```

### Steps

1. Run curl with Daily granularity and no errrors should be returned:
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/tally/products/RHEL%20Server/Cores?granularity=Daily&beginning=2023-07-21T17%3A32%3A28Z&ending=2023-07-21T17%3A32%3A28Z&use_running_totals_format=false' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
1. Run curl with Hourly granularity and unsupported granularity error should return:
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/tally/products/RHEL%20Server/Cores?granularity=Hourly&beginning=2023-07-21T17%3A32%3A28Z&ending=2023-07-21T17%3A32%3A28Z&use_running_totals_format=false' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```

### CODE REVIEW

**Comment by kflahert** [comment]
@ssidhaye Rhel Desktop is no longer a valid productID after the switch to product-config. Can you update the IQE tests to remove references to it? I also just updated the spec to remove that ID.

**Comment by ssidhaye** [comment]
The test account has been opted in 

```
In [22]: app.rhsm_subscriptions.get_tally_report(product_id="RHEL for x86", beginning
    ...: ="2023-08-20T07:00Z", ending="2023-08-20T07:00Z", granularity="Daily")
2023-08-24 08:42:08.951 [    INFO] [root] REST: http://swatch-api-service.ephemeral-q0klpm.svc:8000/api/rhsm-subscriptions/v1/tally/products/RHEL%20for%20x86
2023-08-24 08:42:08.973 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=None, params=[('granularity', 'Daily'), ('beginning', '2023-08-20T07:00:00+00:00'), ('ending', '2023-08-20T07:00:00+00:00')]
Out[22]: 
{'data': [{'cloud_cores': 0,
           'cloud_instance_count': 0,
           'cloud_sockets': 0,
           'core_hours': 0.0,
           'cores': 0,
           'date': datetime.datetime(2023, 8, 20, 0, 0, tzinfo=tzlocal()),
           'has_data': False,
           'hypervisor_cores': 0,
           'hypervisor_instance_count': 0,
           'hypervisor_sockets': 0,
           'instance_count': 0,
           'instance_hours': 0.0,
           'physical_cores': 0,
           'physical_instance_count': 0,
           'physical_sockets': 0,
           'sockets': 0}],
 'meta': {'count': 1,
          'granularity': 'Daily',
          'product': 'RHEL for x86',
          'total_core_hours': 0.0,
          'total_instance_hours': 0.0}}

In [23]: app.rhsm_subscriptions.get_tally_report(product_id="RHEL for x86", beginning
    ...: ="2023-08-20T07:00Z", ending="2023-08-20T07:00Z", granularity="Hourly")
2023-08-24 08:42:43.431 [    INFO] [root] REST: http://swatch-api-service.ephemeral-q0klpm.svc:8000/api/rhsm-subscriptions/v1/tally/products/RHEL%20for%20x86
2023-08-24 08:42:43.453 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=None, params=[('granularity', 'Hourly'), ('beginning', '2023-08-20T07:00:00+00:00'), ('ending', '2023-08-20T07:00:00+00:00')]
2023-08-24 08:42:43.453 [   ERROR] [iqe_rhsm_subscriptions] (400)
Reason: 
HTTP response headers: HTTPHeaderDict({'Set-Cookie': 'JSESSIONID=01402D54E93824B308B317E902C2ED78; Path=/; HttpOnly', 'X-Content-Type-Options': 'nosniff', 'X-XSS-Protection': '0', 'Cache-Control': 'no-cache, no-store, max-age=0, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Frame-Options': 'DENY', 'Content-Type': 'application/vnd.api+json', 'Content-Length': '130', 'Date': 'Thu, 24 Aug 2023 08:42:43 GMT', 'Connection': 'close'})
HTTP response body: {"errors":[{"status":"400","code":"SUBSCRIPTIONS1001","title":"RHEL for x86 does not support any granularity finer than Hourly"}]}

Out[23]: False
```

The changes LGTM based on above observations.

There is one failure in the tests which Aurobinda has fixed, I'll retrigger the test again in the evening to get a green run

**Comment by ssidhaye** [comment]
@kflahert I don't see RHEL Desktop in the product list. I think @lindseyburnett  updated those

**Comment by lindseyburnett** [comment]
> @kflahert I don't see RHEL Desktop in the product list. I think @lindseyburnett updated those

Correct, RHEL Desktop no longer exists

File: swatch-product-configuration/src/test/java/com/redhat/swatch/configuration/registry/VariantTest.java [hunk 1]
```diff
@@ -58,4 +58,16 @@ void testFindByEngineeringId() {
 
     assertEquals("RHEL for x86", actual.get().getTag());
   }
+
+  @Test
+  void testGranularityCompatibility() {
+    assertTrue(
+        Variant.granularityIsCompatible("RHEL SERVER", SubscriptionDefinitionGranularity.DAILY));
```
Comment (lindseyburnett) [inline swatch-product-configuration/src/test/java/com/redhat/swatch/configuration/registry/VariantTest.java:65]: RHEL Server isn't a thing anymore.  **RHEL Server** became **RHEL for x86**

File: swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Variant.java [hunk 2]
```diff
@@ -78,4 +78,13 @@ public static Optional<Variant> findByTag(String defaultVariantTag) {
         .filter(variant -> Objects.equals(variant.getTag(), defaultVariantTag))
         .findFirst();
   }
+
+  public static boolean granularityIsCompatible(
```
Comment (lindseyburnett) [inline swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Variant.java:82]: Do you mind renaming to `isGranularityCompatible`?

**Review by lindseyburnett — CHANGES_REQUESTED** [review]

**Review by lindseyburnett — APPROVED** [review]

**Review by ssidhaye — APPROVED** [review]
Based on test observations changes LGTM

**Pull request merged**